### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by stevehodgkiss**
More technical and specific about why you might want to use StackMaster to manage CF stacks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 
-- See the changes you are making to a stack before you apply them
-- Connect stacks
-- Keep secrets secret
-- Customise stacks in different environments
-- Apply descriptive labels to regions
+- Synchronous visibility into stack updates. See exactly what is changing and
+  what will happen before agreeing to apply a change.
+- Dynamic paramter resolvers.
+- Template compiler support for YAML and SparkleFormation.
```
Comment (taoza) [inline README.md]: Maybe keep the link to sparkle as you had before: `[SparkleFormation](http://www.sparkleformation.io)`

File: README.md [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
-StackMaster provides an easy command line interface to managing CloudFormation
-stacks defined with templates specified in either the
-[SparkleFormation](http://www.sparkleformation.io) DSL or standard JSON format.
+Stack updates can cause a lot of damage if applied blindly. StackMaster helps
+with this by providing the operator with as much information about the proposed
+change as possible before asking for confirmation to continue. That information
+includes:
+
+- Template body and parameter diffs
+- Change sets are displayed for review before stack updates can be applied
```
Comment (taoza) [inline README.md]: Maybe link to AWS's change sets doc since it's a new concept at the moment.

File: README.md [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+- Once the diffs & change set look okay, the change can be confirmed to start
+  the update and watch stack events
+
+A stacks parameters can be dynamically resolved at runtime using one of the
```
Comment (taoza) [inline README.md]: `s/A stacks parameters/Stack parameters/`?

File: README.md [hunk 4]
```diff
@@ -1,17 +1,28 @@
...
+built in parameter resolvers. Parameters can be sourced from GPG files, other
```
Comment (taoza) [inline README.md]: `s/GPG files/GPG secret files/`?

File: README.md [hunk 5]
```diff
@@ -1,17 +1,28 @@
...
+stacks outputs, querying various AWS API's to get resource ARNs etc.
+
+If you hate writing JSON (we do) templates can alternatively written in YAML or
```
Comment (taoza) [inline README.md]: `s/can alternatively written/can be alternatively written/`?

**Comment by taoza** [comment]
:+1:

File: README.md [hunk 6]
```diff
@@ -1,17 +1,29 @@
...
-- Apply descriptive labels to regions
-
-StackMaster provides an easy command line interface to managing CloudFormation
-stacks defined with templates specified in either the
-[SparkleFormation](http://www.sparkleformation.io) DSL or standard JSON format.
+StackMaster is a CLI tool to manage CloudFormation stacks, with the following features:
+
+- Synchronous visibility into stack updates. See exactly what is changing and
+  what will happen before agreeing to apply a change.
+- Dynamic paramter resolvers.
```
Comment (nemski) [inline README.md]: parameter

Comment (stevehodgkiss) [inline README.md]: Pushed a correction to master :)

File: README.md [hunk 7]
```diff
@@ -1,17 +1,29 @@
...
+change as possible before asking for confirmation to continue. That information
+includes:
+
+- Template body and parameter diffs.
+- [Change
+  sets](https://aws.amazon.com/blogs/aws/new-change-sets-for-aws-cloudformation/)
+are displayed for review.
+- Once the diffs & change set have been reviewed, the change can be applied and
+  stack events monitored.
+- Stack events will be outputted until an end state is reached.
```
Comment (andrewjhumphrey) [inline README.md]: Is outputted a word?  I would have used output, but then I don't english very well so please do ignore me if I'm wrong.

File: README.md [hunk 8]
```diff
@@ -1,17 +1,29 @@
...
+
+Stack parameters can be dynamically resolved at runtime using one of the
+built in parameter resolvers. Parameters can be sourced from GPG encrypted YAML
+files, other stacks outputs, querying various AWS API's to get resource ARNs
```
Comment (andrewjhumphrey) [inline README.md]: by querying (apologies for the subediting)

**Comment by andrewjhumphrey** [comment]
Sorry, just noticed the merged tag, nothing I've said is critical do feel free to ignore.

**Comment by stevehodgkiss** [comment]
@andrewjhumphrey I've addressed your comments in c84b640, thanks!

**Pull request merged**